source/common/http/filter/fault_filter.cc

```diff
@@ -29,48 +29,31 @@ const std::string FaultFilter::ABORT_PERCENT_KEY = "fault.http.abort.abort_perce
 const std::string FaultFilter::DELAY_DURATION_KEY = "fault.http.delay.fixed_duration_ms";
 const std::string FaultFilter::ABORT_HTTP_STATUS_KEY = "fault.http.abort.http_status";
 
-FaultFilterConfig::FaultFilterConfig(const Json::Object& json_config, Runtime::Loader& runtime,
-                                     const std::string& stats_prefix, Stats::Scope& scope)
+FaultFilterConfig::FaultFilterConfig(const envoy::api::v2::filter::http::HTTPFault& fault,
+                                     Runtime::Loader& runtime, const std::string& stats_prefix,
+                                     Stats::Scope& scope)
     : runtime_(runtime), stats_(generateStats(stats_prefix, scope)), stats_prefix_(stats_prefix),
       scope_(scope) {
 
-  json_config.validateSchema(Json::Schema::FAULT_HTTP_FILTER_SCHEMA);
-
-  const Json::ObjectSharedPtr config_abort = json_config.getObject("abort", true);
-  const Json::ObjectSharedPtr config_delay = json_config.getObject("delay", true);
-
-  if (config_abort->empty() && config_delay->empty()) {
-    throw EnvoyException("fault filter must have at least abort or delay specified in the config.");
-  }
-
-  if (!config_abort->empty()) {
-    abort_percent_ = static_cast<uint64_t>(config_abort->getInteger("abort_percent", 0));
-
-    // TODO(mattklein123): Throw error if invalid return code is provided
-    http_status_ = static_cast<uint64_t>(config_abort->getInteger("http_status"));
+  if (fault.has_abort()) {
+    abort_percent_ = fault.abort().percent();
+    http_status_ = fault.abort().http_status();
   }
 
-  if (!config_delay->empty()) {
-    const std::string type = config_delay->getString("type");
-    ASSERT(type == "fixed");
-    UNREFERENCED_PARAMETER(type);
-    fixed_delay_percent_ =
-        static_cast<uint64_t>(config_delay->getInteger("fixed_delay_percent", 0));
-    fixed_duration_ms_ = static_cast<uint64_t>(config_delay->getInteger("fixed_duration_ms", 0));
+  if (fault.has_delay()) {
+    fixed_delay_percent_ = fault.delay().percent();
+    auto delay = fault.delay();
```

Comment:
Nit: `const auto& delay` (or just inline below)

Reply:
inlining doesn't work. something is complaining